### PR TITLE
Engine - Improve Test Coverage for router Integration Tests

### PR DESCRIPTION
**Pull request by NahuFigueroa97**
|Related issue|
|---|
|#21758|

## Description

The purpose of this enhancement is to implement the use of Behave to perform extensive integration tests on the Router API. These tests will cover various user stories and scenarios to ensure the proper functioning of the API. In addition, they will be extended to cover tests on the KVDB command line interface (CLI) and the router API.

### CODE REVIEW

File: src/engine/test/integration_tests/policy/steps/api_steps.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+api_client = APIClient(SOCKET_PATH)
+
+def run_command(command):
+    result = subprocess.run(
+        command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    assert result.returncode == 0, f"{result.stderr}"
+
+def send_recv(request, expected_response_type) -> Tuple[Optional[str], dict]:
+    error, response = api_client.send_recv(request)
+    parse_response = ParseDict(response, expected_response_type)
```
Comment (juliancnn) [inline src/engine/test/integration_tests/policy/steps/api_steps.py:24]: Check `error`, this var handle the protocol error:

https://github.com/wazuh/wazuh/blob/aeac12615a30969ac9142da37e65ea2368b03435/src/engine/tools/api-communication/src/api_communication/client.py#L137-L139

**Review by juliancnn — CHANGES_REQUESTED** [review]

**Review by JavierBejMen — APPROVED** [review]
LGTM!

**Review by JavierBejMen — APPROVED** [review]
LGTM

**Review by JcabreraC — APPROVED** [review]
LGTM !

**Pull request merged**